dev/TabView/TabView.cpp

```diff
@@ -1232,18 +1297,131 @@ int TabView::GetItemCount()
+    //
+    //    Tab 1 -> Tab 1 close button -> Tab 2 -> Tab 2 close button -> ... -> Tab N -> Tab N close button -> Add tab button -> Tab 1
+    //
+    // Any element that's not focusable is skipped.
+    //
+    std::vector<winrt::Control> focusOrderList;
+
+    for (int i = 0; i < GetItemCount(); i++)
+    {
+        if (auto tab = ContainerFromIndex(i).try_as<winrt::TabViewItem>())
+        {
+            if (SharedHelpers::IsFocusable(tab, false /* checkTabStop */))
+            {
+                focusOrderList.push_back(tab);
+
+                if (auto closeButton = winrt::get_self<TabViewItem>(tab)->GetCloseButton())
+                {
+                    if (SharedHelpers::IsFocusable(closeButton, false /* checkTabStop */))
+                    {
+                        focusOrderList.push_back(closeButton);
+                    }
+                }
+            }
+        }
+    }
+
+    if (auto&& addButton = m_addButton.get())
+    {
+        if (SharedHelpers::IsFocusable(addButton, false /* checkTabStop */))
+        {
+            focusOrderList.push_back(addButton);
+        }
+    }
+
+    auto position = std::find(focusOrderList.begin(), focusOrderList.end(), focusedControl);
+
+    // The focused control is not in the focus order list - nothing for us to do here either.
+    if (position == focusOrderList.end())
+    {
+        return false;
+    }
+
+    // At this point, we know that the focused control is indeed in the focus list, so we'll move focus to the next or previous control in the list.
+
+    int sourceIndex = static_cast<int>(position - focusOrderList.begin());
+    const int listSize = static_cast<int>(focusOrderList.size());
+    const int increment = moveForward ? 1 : -1;
+    int currentIndex = sourceIndex + increment;
+
+    while (currentIndex != sourceIndex)
+    {
+        if (currentIndex < 0)
+        {
+            currentIndex = listSize - 1;
+        }
+        else if (currentIndex >= listSize)
+        {
+            currentIndex = 0;
+        }
+
+        // We have to do a bit of a dance for the close buttons - we don't want users to be able to give them focus when tabbing through an app,
+        // since we only want to tab into the TabView once and then tab out on the next tab press.  However, IsTabStop also controls keyboard
+        // focusability in general - we can't give keyboard focus to a control with IsTabStop = false.  To work around this, we'll temporarily set
+        // IsTabStop = true before calling Focus(), and then set it back to false if it was previously false.
+
+        auto&& control = focusOrderList[currentIndex];
+        const bool originalIsTabStop = control.IsTabStop();
+
+        auto scopeGuard = gsl::finally([control, originalIsTabStop]()
+            {
+                control.IsTabStop(originalIsTabStop);
+            });
+
+        control.IsTabStop(true);
+
+        if (control.Focus(winrt::FocusState::Keyboard))
+        {
+            return true;
```

Comment:
Once you focus the control, won't IsTabStop get immediately set to false? Does that cause the control to lose focus? Or can it stay focusable even if it is not a tab stop?

Reply:
It can stay focused even if it's not a tab stop; it just can't take keyboard focus in that case.

Reply:
Interesting. Thanks.